### PR TITLE
Fix Authentication issues in ForceOauthClient and ForceSoapValidator

### PR DESCRIPTION
**Pull request by asl-acx**

### CODE REVIEW

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java [hunk 1]
```diff
@@ -85,31 +101,46 @@ private static void extractPartnerUrl(ForceUserInfo userInfo) {
         userInfo.setPartnerUrl(userInfo.getUrls().get("partner").replace("{version}", API_VERSION));
     }
 
-    private static boolean isBadTokenError(HttpResponseException e) {
-        return e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN;
+    private boolean isBadTokenError(HttpResponseException e) {
+        return ((e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN)
+                &&
+                StringUtils.equalsAnyIgnoreCase(responseContent, BAD_TOKEN_SF_ERROR_CODE) ||
```
Comment (gorchenok) [inline sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: Please use StringUtils.equalsAnyIgnoreCase() in correct way StringUtils.equalsAnyIgnoreCase(responseContent,  BAD_TOKEN_SF_ERROR_CODE,  MISSING_TOKEN_SF_ERROR_CODE, WRONG_ORG_SF_ERROR_CODE)

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java [hunk 2]
```diff
@@ -85,31 +101,46 @@ private static void extractPartnerUrl(ForceUserInfo userInfo) {
...
+                StringUtils.equalsAnyIgnoreCase(responseContent, MISSING_TOKEN_SF_ERROR_CODE) ||
+                StringUtils.equalsAnyIgnoreCase(responseContent, WRONG_ORG_SF_ERROR_CODE))
+                ||
+                (e.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND &&
+                        StringUtils.equalsAnyIgnoreCase(responseContent, BAD_ID_SF_ERROR_CODE));
```
Comment (gorchenok) [inline sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: Please use equalsIgnoreCase() for single constant

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java [hunk 3]
```diff
@@ -85,31 +101,46 @@ private static void extractPartnerUrl(ForceUserInfo userInfo) {
...
-    private static boolean isInternalError(HttpResponse response) {
-        try {
-            return response.getStatusCode() / 100 == 5 ||
-                    (response.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND
-                            && response.getContent().equals(INTERNAL_SERVER_ERROR_SF_ERROR_CODE));
+    private boolean isInternalError(HttpResponse response) {
+        try (InputStream is = response.getContent()) {
+            responseContent = IOUtils.toString(response.getContent(), StandardCharsets.UTF_8);
         } catch (IOException e) {
             return false;
```
Comment (gorchenok) [inline sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: Why we don't need retry if got IOException?

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java (hunk 1 above)
Comment (asl-acx) [reply sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: fixed

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java (hunk 2 above)
Comment (asl-acx) [reply sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: fixed

File: sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java (hunk 3 above)
Comment (asl-acx) [reply sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java]: fixed

**Pull request merged**